### DISCUSSION TITLE
How to listen to router events using the new app router?

### DISCUSSION THREAD

**Discussion by fobos531** [vercel/next.js#52568] [App Router] [answered]
I saw in the docs that the new router import doesn't support this yet - is it however possible to somehow use a "compatibility layer" of sorts to enable this functionality in the app router

**Comment by icyJoseph** [comment, marked as answer]
There's no such thing at the moment, other than hooks that you can try to use to listen to route changes. 

- https://github.com/vercel/next.js/discussions/41745#discussioncomment-3987025
- https://github.com/vercel/next.js/discussions/42016#discussioncomment-3991279

However it is nothing like, route change started, route change completed, route change errored...
